codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/CustomizableOperationDecorator.kt

```diff
@@ -5,12 +5,9 @@
 
 package software.amazon.smithy.rust.codegen.client.smithy.generators.client
 
-import software.amazon.smithy.rust.codegen.core.smithy.customize.NamedCustomization
-import software.amazon.smithy.rust.codegen.core.smithy.customize.Section
+import software.amazon.smithy.rust.codegen.core.smithy.customize.AdHocSection
 
-sealed class CustomizableOperationSection(name: String) : Section(name) {
+sealed class CustomizableOperationSection(name: String) : AdHocSection(name) {
```

Comment:
You changed this to adhoc, but didn't actually make any calls to adhocCustomization anywhere. Should we just remove this entirely? I could also be misunderstanding how these work again... they're not very clear.

Reply:
there is a call to adhocCustomizations here:
https://github.com/smithy-lang/smithy-rs/blob/627a1a4cb76cdf4b955535c0fe03bf7e329143bb/aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsPresigningDecorator.kt#L151